Tests/Integration/inc/classes/ImagifySettings/updateSiteOptionOnNetwork.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Imagify\Tests\Integration\inc\classes\ImagifySettings;
+
+use Imagify_Settings;
+use Imagify\Tests\Integration\TestCase;
+use Brain\Monkey\Functions;
+
+/**
+ * @covers Imagify_Settings::update_site_option_on_network
+ *
+ * @uses   imagify_check_nonce()
+ * @uses   imagify_die()
+ * @uses   imagify_maybe_redirect()
+ *
+ * @group  ImagifySettings
+ */
+class Test_UpdateSiteOptionOnNetwork extends TestCase {
+
+	private $user_id;
+
+	public function provideData() {
+		return $this->getTestData( __DIR__, 'updateSiteOptionOnNetwork' );
+	}
+
+	/**
+	 * @dataProvider provideData
+	 */
+	public function testShouldUpdateNetworkSettings( $config, $expected ) {
+		$_POST['option_page'] = $config['option_page'];
+
+		if ( $config['user_can'] ) {
+			$this->user_id = $this->factory->user->create( [ 'role' => 'administrator', ] );
+			$admin         = get_role( 'administrator' );
+			$admin->add_cap( 'manage_network_options' );
```

Comment:
To prevent future conflicts between tests, you will want to reset the capability, & unset the `option_page` key on tearDown.

This can be applied to anything you set in the test method that is for a specific state, and which might not be accurate for another test.

Reply:
Done